docs/mine-hnt/validators/validator-deployment-guide.mdx

```diff
@@ -0,0 +1,402 @@
+* `--name validator` names the container, which makes interacting with the docker easier, but feel free to name the container whatever you want.
+* `--mount` the parameters above will mount the container's `/var/data/` directory to the systems directory `$HOME/validator_data`. You will want this folder to be persistent across runs of the docker container as it will contain both the blockchain data and the miner key of your validator.
+
+
+#### Upgrade your Docker container
+
+We anticipate numerous, frequent updates during Testnet and recommend using the [latest miner tag found here](https://quay.io/repository/team-helium/validator?tag=latest&tab=tags).
+
+To upgrade your docker container to the latest version:
+
+```
+docker stop validator && docker rm validator
+```
+
+Then re-launch docker with the same command as above with the updated version tag.
+
+#### Interact with the Validator within the Container
+
+You may want to interrogate the Validator or interact with it. Docker's `exec` command enables this. For example:
+
+```
+docker exec validator miner info height
+```
+
+
+As shown above, you can prepend `docker exec validator` to any of the commands documented or create an alias such as
+
+```
+alias validator="docker exec validator miner"
+```
+
+And start the container again as described above, but with the new release tag.
+
+And thanks to the `--mount option`, the blockchain data and the validator keys are preserved through updates.
+
+:::info
+
+Now that you have your Validator running, you need to [actually Stake tokens to make it official.](#stake-tokens-to-your-validator)
+
+:::
+
+
+### Deploy Miner from Source 
+
+Ok, brave soul. Here's what you'll need to build and deploy the miner from source:
+
+
+* miner
+* Erlang package (the latest 22 or 23 versions are supported, the 24 release candidate will NOT work)
+* `libwxgtk` package
+* git 
+
+
+Before we get started, make sure you're fully up to date:
+
+```
+sudo apt-get update -y
+```
+
+Next, let's install all the things. Start by cloning the `validator` branch of `miner`:
+
+```
+git clone -b pevm/validators https://github.com/helium/miner.git
+```
+
+Then proceed to get you some Erlang (here we're specifying `22.3.1`):
+
+```
+wget https://packages.erlang-solutions.com/erlang/debian/pool/esl-erlang_22.3.1-1~ubuntu~bionic_amd64.deb
+```
+
+Next, get the `libwxgtk` package:
+
+```
+wget http://mirrors.kernel.org/ubuntu/pool/universe/w/wxwidgets3.0/libwxgtk3.0-0v5_3.0.4+dfsg-3_amd64.deb
+```
+
+And finish thins off by acquiring a some wonderful dependencies:
+
+```
+sudo apt install -y libdbus-1-dev autoconf automake libtool flex libgmp-dev cmake libsodium-dev libssl-dev bison libsnappy-dev libclang-dev doxygen make cargo
+```
+
+```
+sudo apt install g++ -y
```

Comment:
is there a reason these need to be installed individually on their own line? or could the be added to the mega-apt one-liner above?

Reply:
I don't see why we can't add them. I'll update.